### ISSUE TITLE
Rule proposal/modification: [naming-convention] Support modifier for `as const` in naming convention rule

### ISSUE THREAD

**Issue by AllySummers** [typescript-eslint/typescript-eslint#4698]
### Description

I believe this would be beneficial in places where you wish to enforce (or allow) a different naming convention for something that is marked `as const` (i.e. make it behave similarly to an enum).

For example, while there are some small differences, the following 2 declarations are very similar:
```ts
export enum Direction {
  Up = 'UP',
  Down = 'DOWN',
  Left = 'LEFT',
  Right = 'RIGHT'
}

export const Direction = {
  Up: 'UP',
  Down: 'DOWN',
  Left: 'LEFT',
  Right: 'RIGHT'
} as const;
```

For this reason, we may enforce that variables must either be `snake_case` or `UPPER_CASE`, but in the event that something uses `as const`, we would like to allow `PascalCase`

### Fail

(N/A?)

### Pass

(N/A?)

**Comment by bradzacher** [comment]
Thanks for the suggestion!

After consideration - I think that this use-case is probably too narrow for it to be considered for inclusion in this general-purpose plugin.
